code/forms/JsonField.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/**
+ * Class JsonField combines form inputs into a key-value pair
+ */
+class JsonField extends CompositeField
+{
+    public function __construct($name, $children = null)
+    {
+        $this->setName($name);
+
+        if ($children instanceof FieldList) {
+            foreach ($children as $child) {
+                $child->setName($this->getName() . "[" . $child->getName() . "]");
+            }
+        } elseif (is_array($children)) {
+            foreach ($children as $child) {
+                $child->setName($this->getName() . "[" . $child->getName() . "]");
+            }
+        } else {
+            $children = is_array(func_get_args()) ? func_get_args() : array();
+            if (!empty($children)) {
+                array_shift($children);
+            }
+            foreach ($children as $child) {
+                $child->setName($this->getName() . "[" . $child->getName() . "]");
+            }
+        }
+        parent::__construct($children);
+    }
+
+    public function hasData()
+    {
+        return true;
+    }
+
+    /**
+     * Have to has it to overwritten CompositeField behavior
+     *
+     * @param      $list
+     * @param bool $saveableOnly
+     */
+
+    public function collateDataFields(&$list, $saveableOnly = false)
+    {
+    }
+
+    function array_filter_empty_recursive($haystack)
```

Comment:
* Add method visibility
* Name with camel case
* Add doc block explaining what it's for and a test (if haven't already)